src/org/omegat/core/dictionaries/DictionaryEntry.java

```diff
@@ -49,6 +62,10 @@ public String getArticle() {
 
     @Override
     public int compareTo(DictionaryEntry o) {
-        return word.compareTo(o.word);
+        if (query.equals(o.query)) {
+            return word.compareTo(o.word);
+        } else {
+            return query.compareTo(o.query);
+        }
```

Comment:
You could slightly optimize this like

```suggestion
        int queryComparison = query.compareTo(o.query);
        if (queryComparison != 0) {
            return queryComparison;
        } else {
            return word.compareTo(o.word);
        }
```

Reply:
accepted the suggestion.